src/control/build/interop.go

```diff
@@ -103,6 +114,25 @@ var defaultRules = []InteropRule{
 	},
 }
 
+var (
+	Server22xAgent20x = &InteropRule{
+		Self:          ComponentServer,
+		Other:         ComponentAgent,
+		Description:   "server v2.2.x is compatible with agent v2.0.x",
+		StopOnSuccess: true,
+		Check: func(self, other *VersionedComponent) bool {
+			// We assume that an unversioned agent is 2.0.x.
+			v0_0_0 := MustNewVersion("0.0.0")
+			v2_2_0 := MustNewVersion("2.1.100")
+			v2_3_0 := MustNewVersion("2.3.0")
+			return (self.Version.GreaterThanOrEquals(v2_2_0) &&
+				self.Version.LessThan(v2_3_0)) &&
```

Comment:
nit: logic would be clearer to me with this on the previous line (and the next 2 lines on the same line)